src/ChatClient.ts

```diff
@@ -144,23 +144,32 @@ export class ChatClient {
   }
 
   async join() {
-    const {email, channelId, teamId, channelName} = await this.sendCommand<any, any>('join', {
-      version: getAppVersion(),
-    });
-    if (!email || !channelId || !teamId || !channelName) {
-      console.error(`Client ready without email or channel. Disconnecting...`);
-      this.client.close();
-    }
+    try {
+      const {email, channelId, teamId, channelName} = await this.sendCommand<any, any>('join', {
+        version: getAppVersion(),
+      });
+      if (!email || !channelId || !teamId || !channelName) {
+        console.error(`Client ready without email or channel. Disconnecting...`);
+        this.client.close();
+        this.fireError(new Error('No mapping found.'));
+        return;
```

Comment:
Could it be that the issue with repetitive reconnection happens only due to this forgotten return?

Reply:
Actually, we never go there! The server responds an error, thus the catch is what closes the connection. But if the server would respond a 200 with an error object, we would end up here (this was the error handling I have now removed). Adding the `return` is nicer but currently totally useless.